src/main/java/org/commonjava/util/partyline/impl/infinispan/model/FileBlock.java

```diff
@@ -110,11 +110,11 @@ public ByteBuffer getBuffer()
         return data;
     }
 
-    byte[] getByteArray()
+    private byte[] getByteArray()
     {
         if ( data.hasArray() )
         {
-            return data.array();
+            return data.array(); // TODO: if we use data.array() we might also save the arrayOffset
```

Comment:
This could be expensive...but if we actually use this in the readers then we shouldn't need to worry about maintaining reader-local positions, since we won't be reading subsections of a block...